### PR TITLE
fixing cache invalidation for screenshot updates

### PR DESCRIPTION
**Pull request by martinraison**
@andrewconner @yingjieMiao @raymondng

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala [hunk 1]
```diff
@@ -261,7 +261,9 @@ extends DbRepo[NormalizedURI] with NormalizedURIRepo with ExternalIdColumnDbFunc
   }
 
   def updateScreenshotUpdatedAt(id: Id[NormalizedURI], time: DateTime)(implicit session: RWSession) = {
-    (for {t <- rows if t.id === id} yield (t.updatedAt, t.screenshotUpdatedAt)).update((clock.now, time))
+    val updateTime = clock.now
+    (for {t <- rows if t.id === id} yield (t.updatedAt, t.screenshotUpdatedAt)).update((updateTime, time))
+    invalidateCache(get(id).copy(updatedAt = updateTime, screenshotUpdatedAt = Some(time)))
```
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala:266]: this looks weird but ok given our current cache invalidation model ...

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala [hunk 2]
```diff
@@ -261,7 +261,9 @@ extends DbRepo[NormalizedURI] with NormalizedURIRepo with ExternalIdColumnDbFunc
...
   }
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala:267]: I assume `clock.now` is the same as 'currentDateTime'? mostly time zone concern.

**Comment by raymondng** [comment]
lgtm

**Comment by yingjieMiao** [comment]
lgtm

**Comment by andrewconner** [comment]
As mentioned on hipchat, you may want to increment the cache's version, so it fixes the issue for broken links.

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala (hunk 2 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala:267]: `clock.now` is an injectable form of `currentDateTime`. It's preferred in any case where we can use it.

**Pull request merged**